Tasks/FtpUpload/ftputils.ts

```diff
@@ -82,7 +82,23 @@ export class FtpHelper {
 
         this.ftpClient.rmdir(remotePath, true, function (err) {
             if (err) {
-                defer.reject('Unable to clean remote folder: ' + remotePath + ' error: ' + err);
+                defer.reject('Unable to remove remote folder: ' + remotePath + ' error: ' + err);
+            } else {
+                defer.resolve(null);
+            }
+        });
+
+        return defer.promise;
+    }
+
+    remoteDelete(remotePath: string): Q.Promise<void> {
+        var defer: Q.Deferred<void> = Q.defer<void>();
```

Comment:
let instead of var - throughout the file
const would also work in many locations (I prefer const when you can, but it doesn't matter that much)